**<파일 경로>**
- C:/Users/Brian/Desktop/Data Science/Projects/Playdata/dataset/population/train

In [1]:
filepath = input()

C:/Users/Brian/Desktop/Data Science/Projects/Playdata/dataset/population/train


In [2]:
import numpy as np
import pandas as pd
from tqdm.notebook import tqdm

# 2018~2019년 인구 데이터 결합
**<참고>**
- 데이터들을 불러올 때, 다음과 같은 warning이 발생하였다.
    - DtypeWarning: Columns (1) have mixed types.Specify dtype option on import or set low_memory=False.
    - 이를 해결해주기 위해 ```dtype = {'기준일ID': str}``` 라는 옵션을 걸어주었다.
        - 여기서는 Columns (1) = '기준일ID'이다.

In [3]:
# 2018년 인구 데이터
file_1 = pd.read_csv('{}/LOCAL_PEOPLE_DONG_201801.csv'.format(filepath), dtype = {'기준일ID': str})
file_2 = pd.read_csv('{}/LOCAL_PEOPLE_DONG_201802.csv'.format(filepath), dtype = {'기준일ID': str})
file_3 = pd.read_csv('{}/LOCAL_PEOPLE_DONG_201803.csv'.format(filepath), dtype = {'기준일ID': str})
file_4 = pd.read_csv('{}/LOCAL_PEOPLE_DONG_201804.csv'.format(filepath), dtype = {'기준일ID': str})
file_5 = pd.read_csv('{}/LOCAL_PEOPLE_DONG_201805.csv'.format(filepath), dtype = {'기준일ID': str})
file_6 = pd.read_csv('{}/LOCAL_PEOPLE_DONG_201806.csv'.format(filepath), dtype = {'기준일ID': str})
file_7 = pd.read_csv('{}/LOCAL_PEOPLE_DONG_201807.csv'.format(filepath), dtype = {'기준일ID': str})
file_8 = pd.read_csv('{}/LOCAL_PEOPLE_DONG_201808.csv'.format(filepath), dtype = {'기준일ID': str})
file_9 = pd.read_csv('{}/LOCAL_PEOPLE_DONG_201809.csv'.format(filepath), dtype = {'기준일ID': str})
file_10 = pd.read_csv('{}/LOCAL_PEOPLE_DONG_201810.csv'.format(filepath), dtype = {'기준일ID': str})
file_11 = pd.read_csv('{}/LOCAL_PEOPLE_DONG_201811.csv'.format(filepath), dtype = {'기준일ID': str})
file_12 = pd.read_csv('{}/LOCAL_PEOPLE_DONG_201812.csv'.format(filepath), dtype = {'기준일ID': str})

# 2019년 인구 데이터
file_13 = pd.read_csv('{}/LOCAL_PEOPLE_DONG_201901.csv'.format(filepath), dtype = {'기준일ID': str})
file_14 = pd.read_csv('{}/LOCAL_PEOPLE_DONG_201902.csv'.format(filepath), dtype = {'기준일ID': str})
file_15 = pd.read_csv('{}/LOCAL_PEOPLE_DONG_201903.csv'.format(filepath), dtype = {'기준일ID': str})
file_16 = pd.read_csv('{}/LOCAL_PEOPLE_DONG_201904.csv'.format(filepath), dtype = {'기준일ID': str})
file_17 = pd.read_csv('{}/LOCAL_PEOPLE_DONG_201905.csv'.format(filepath), dtype = {'기준일ID': str})
file_18 = pd.read_csv('{}/LOCAL_PEOPLE_DONG_201906.csv'.format(filepath), dtype = {'기준일ID': str})
file_19 = pd.read_csv('{}/LOCAL_PEOPLE_DONG_201907.csv'.format(filepath), dtype = {'기준일ID': str})
file_20 = pd.read_csv('{}/LOCAL_PEOPLE_DONG_201908.csv'.format(filepath), dtype = {'기준일ID': str})
file_21 = pd.read_csv('{}/LOCAL_PEOPLE_DONG_201909.csv'.format(filepath), dtype = {'기준일ID': str})
file_22 = pd.read_csv('{}/LOCAL_PEOPLE_DONG_201910.csv'.format(filepath), dtype = {'기준일ID': str})
file_23 = pd.read_csv('{}/LOCAL_PEOPLE_DONG_201911.csv'.format(filepath), dtype = {'기준일ID': str})
file_24 = pd.read_csv('{}/LOCAL_PEOPLE_DONG_201912.csv'.format(filepath), dtype = {'기준일ID': str})

In [4]:
concat_list = [file_1, file_2, file_3, file_4, file_5, file_6, file_7, file_8, file_9, file_10, file_11, file_12,
               file_13, file_14, file_15, file_16, file_17, file_18, file_19, file_20, file_21, file_22, file_23, file_24]

pop_total = pd.concat(concat_list, axis = 0, ignore_index = True)
pop_total.head()

Unnamed: 0     기준일ID  시간대구분     행정동코드      총생활인구수  남자0세부터9세생활인구수  \
0           0  20180101      0  11110515  14471.4116       527.7090   
1           1  20180101      0  11110530  23059.7992       715.9096   
2           2  20180101      0  11110540   4928.6410        83.2518   
3           3  20180101      0  11110550  14060.9810       268.6168   
4           4  20180101      0  11110560  20322.3298       515.5184   

   남자10세부터14세생활인구수  남자15세부터19세생활인구수  남자20세부터24세생활인구수  남자25세부터29세생활인구수  ...  \
0         336.2573         381.2662         551.7157         567.6465  ...   
1         318.1821         461.8743        1074.8617         959.5067  ...   
2          63.7905          86.3778         255.9185         253.8867  ...   
3         208.2819         331.6045         483.0823         421.1730  ...   
4         372.8914         552.9593         594.7631         559.0723  ...   

   여자25세부터29세생활인구수  여자30세부터34세생활인구수  여자35세부터39세생활인구수  여자40세부터44세생활인구수  \
0         389.1351         473.6825         635.6879         729.5227   
1        1055.1869         975.7669        1025.8741         957.0444   
2         172.6411         142.7509         191.2020         197.6714   
3         384.2033         413.4419         545.6471         511.8307   
4         513.4737         568.1072         711.6436         706.7804   

   여자45세부터49세생활인구수  여자50세부터54세생활인구수  여자55세부터59세생활인구수  여자60세부터64세생활인구수  \
0         610.9207         575.3451         494.6534         463.5910   
1         903.3613         778.7224         774.6881         666.7125   
2         189.4098         168.4414         223.6928         191.1262   
3         701.3981         679.1825         571.2549         629.6860   
4         892.6337         969.5679         982.7466         872.2911   

   여자65세부터69세생활인구수  여자70세이상생활인구수  
0         449.7000     1351.0037  
1         453.2184     2046.0369  
2         128.7450      491.3451  
3         462.4967     1543.6116  
4         658.0842     2322.3739  

[5 rows x 33 columns]

In [5]:
pop_total.drop(['Unnamed: 0'], axis = 1, inplace = True)
pop_total.tail()

기준일ID  시간대구분     행정동코드      총생활인구수  남자0세부터9세생활인구수  \
7306363  20191231     23  11740610  47108.0384      1530.5533   
7306364  20191231     23  11740685  56004.5812      1777.7481   
7306365  20191231     23  11740640  26471.4234      1150.2747   
7306366  20191231     23  11740650  27986.5462      1070.9042   
7306367  20191231     23  11740700  32974.4339      1364.7863   

         남자10세부터14세생활인구수  남자15세부터19세생활인구수  남자20세부터24세생활인구수  남자25세부터29세생활인구수  \
7306363         684.2480        1261.2936        1346.8427        1772.9466   
7306364         859.6586        1252.0765        1500.0176        1999.8669   
7306365         594.0026         731.9921         543.0908         682.9012   
7306366         371.4830         566.1300         747.7386        1173.3330   
7306367         688.4728         816.0758         762.3534         957.2146   

         남자30세부터34세생활인구수  ...  여자25세부터29세생활인구수  여자30세부터34세생활인구수  \
7306363        1919.2425  ...        1778.4421        1836.2225   
7306364        1869.7091  ...        1837.8152        2173.7660   
7306365         856.1581  ...         809.2766         926.6463   
7306366        1189.2734  ...        1246.6635        1167.0023   
7306367         986.8085  ...        1005.1095         951.7186   

         여자35세부터39세생활인구수  여자40세부터44세생활인구수  여자45세부터49세생활인구수  여자50세부터54세생활인구수  \
7306363        1842.3955        1669.1141        1746.4848        1805.8380   
7306364        2183.6664        2088.3162        2559.8771        2309.2951   
7306365        1231.7519        1213.1097        1152.5178        1060.8198   
7306366        1150.3829        1146.3252        1050.6582         997.6744   
7306367        1220.0583        1189.0528        1298.7248        1331.3742   

         여자55세부터59세생활인구수  여자60세부터64세생활인구수  여자65세부터69세생활인구수  여자70세이상생활인구수  
7306363        2198.6277        1971.4508        1509.8941     3397.6009  
7306364        2736.9112        2354.2835        1820.0325     3858.1011  
7306365        1234.5424        1112.0217         937.4364     1743.6741  
7306366        1322.4171        1105.3542         965.8775     1934.5264  
7306367        1598.7414        1458.6600        1275.6589     2708.7187  

[5 rows x 32 columns]

# '기준일ID' 변수 수정
- 데이터를 불러와서 DtypeWarning이 발생한 원인에 대해 확인해보니, 인구 데이터의 "날짜(date)" 변수에 ' 20181130**"** ' 이라는 값이 존재했기 때문임을 확인할 수 있었다.
- 따라서 해당 값을 수정해주겠다.
    - 20181130**"** >> 20181130
- 그 다음 '기준일ID' 변수만 타입을 datetime으로 변경해주겠다.

In [6]:
# 아래의 코드를 통해 index 번호가 '3388608'인 데이터가 문제임을 확인할 수 있다.
pop_total[pop_total['기준일ID'].str.len() != 8]

기준일ID  시간대구분     행정동코드      총생활인구수  남자0세부터9세생활인구수  \
3388608  20181130"      0  11530780  47625.4881      2654.1111   

         남자10세부터14세생활인구수  남자15세부터19세생활인구수  남자20세부터24세생활인구수  남자25세부터29세생활인구수  \
3388608         1046.095        1016.5548        1102.0546        1311.1105   

         남자30세부터34세생활인구수  ...  여자25세부터29세생활인구수  여자30세부터34세생활인구수  \
3388608        1520.9183  ...         1713.402        1731.1882   

         여자35세부터39세생활인구수  여자40세부터44세생활인구수  여자45세부터49세생활인구수  여자50세부터54세생활인구수  \
3388608        2082.0321        2053.8532        1802.4338        1537.2766   

         여자55세부터59세생활인구수  여자60세부터64세생활인구수  여자65세부터69세생활인구수  여자70세이상생활인구수  
3388608        1868.3088        1628.7575         1354.761     3380.1261  

[1 rows x 32 columns]

In [7]:
# 해당 index 번호의 값을 수정
pop_total['기준일ID'][3388608] = '20181130'

<ipython-input-7-e54a2b417de9>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  pop_total['기준일ID'][3388608] = '20181130'


In [8]:
pop_total['기준일ID'] = pd.to_datetime(pop_total['기준일ID'], format = '%Y-%m-%d')

# 변수명 변경
- 한글로 되어 있는 변수명을 모두 영어로 변경해주겠다.

In [9]:
new_colnames = {'기준일ID':'date', '시간대구분':'hour', '행정동코드':'dong_code', '총생활인구수':'total', 
                '남자0세부터9세생활인구수':'m_0_9', '남자10세부터14세생활인구수':'m_10_14', '남자15세부터19세생활인구수':'m_15_19',
                '남자20세부터24세생활인구수':'m_20_24', '남자25세부터29세생활인구수':'m_25_29', '남자30세부터34세생활인구수':'m_30_34',
                '남자35세부터39세생활인구수':'m_35_39', '남자40세부터44세생활인구수':'m_40_44', '남자45세부터49세생활인구수':'m_45_49',
                '남자50세부터54세생활인구수':'m_50_54', '남자55세부터59세생활인구수':'m_55_59', '남자60세부터64세생활인구수':'m_60_64',
                '남자65세부터69세생활인구수':'m_65_69', '남자70세이상생활인구수':'m_over_70', '여자0세부터9세생활인구수':'f_0_9',
                '여자10세부터14세생활인구수':'f_10_14', '여자15세부터19세생활인구수':'f_15_19', '여자20세부터24세생활인구수':'f_20_24',
                '여자25세부터29세생활인구수':'f_25_29', '여자30세부터34세생활인구수':'f_30_34', '여자35세부터39세생활인구수':'f_35_39',
                '여자40세부터44세생활인구수':'f_40_44', '여자45세부터49세생활인구수':'f_45_49', '여자50세부터54세생활인구수':'f_50_54',
                '여자55세부터59세생활인구수':'f_55_59', '여자60세부터64세생활인구수':'f_60_64', '여자65세부터69세생활인구수':'f_65_69',
                '여자70세이상생활인구수':'f_over_70'}

pop_total.rename(columns = new_colnames, inplace = True)
pop_total.head()

date  hour  dong_code       total     m_0_9   m_10_14   m_15_19  \
0 2018-01-01     0   11110515  14471.4116  527.7090  336.2573  381.2662   
1 2018-01-01     0   11110530  23059.7992  715.9096  318.1821  461.8743   
2 2018-01-01     0   11110540   4928.6410   83.2518   63.7905   86.3778   
3 2018-01-01     0   11110550  14060.9810  268.6168  208.2819  331.6045   
4 2018-01-01     0   11110560  20322.3298  515.5184  372.8914  552.9593   

     m_20_24   m_25_29   m_30_34  ...    f_25_29   f_30_34    f_35_39  \
0   551.7157  567.6465  486.6423  ...   389.1351  473.6825   635.6879   
1  1074.8617  959.5067  922.5332  ...  1055.1869  975.7669  1025.8741   
2   255.9185  253.8867  188.0076  ...   172.6411  142.7509   191.2020   
3   483.0823  421.1730  331.6272  ...   384.2033  413.4419   545.6471   
4   594.7631  559.0723  472.6954  ...   513.4737  568.1072   711.6436   

    f_40_44   f_45_49   f_50_54   f_55_59   f_60_64   f_65_69  f_over_70  
0  729.5227  610.9207  575.3451  494.6534  463.5910  449.7000  1351.0037  
1  957.0444  903.3613  778.7224  774.6881  666.7125  453.2184  2046.0369  
2  197.6714  189.4098  168.4414  223.6928  191.1262  128.7450   491.3451  
3  511.8307  701.3981  679.1825  571.2549  629.6860  462.4967  1543.6116  
4  706.7804  892.6337  969.5679  982.7466  872.2911  658.0842  2322.3739  

[5 rows x 32 columns]

# 연령대별 그룹 범주화
- 남/여 구분 없이 연령대 별로 합쳐주겠다.
- 다음과 같이 총 7개의 그룹으로 범주화한 연령대 변수를 생성하겠다.
    - pop_below_20
        - 20세 미만
    - pop_20
        - 20대
    - pop_30
        - 30대
    - pop_40
        - 40대
    - pop_50
        - 50대
    - pop_60
        - 60대
    - pop_over_70
        - 70세 이상

In [10]:
# 총 7개의 그룹으로 범주화한 연령대 변수 생성
pop_total['pop_below_20'] = pop_total['m_0_9'] + pop_total['m_10_14'] + pop_total['m_15_19'] + pop_total['f_0_9'] + pop_total['f_10_14'] + pop_total['f_15_19']
pop_total['pop_20'] = pop_total['m_20_24'] + pop_total['m_25_29'] + pop_total['f_20_24'] + pop_total['f_25_29']
pop_total['pop_30'] = pop_total['m_30_34'] + pop_total['m_35_39'] + pop_total['f_30_34'] + pop_total['f_35_39']
pop_total['pop_40'] = pop_total['m_40_44'] + pop_total['m_45_49'] + pop_total['f_40_44'] + pop_total['f_45_49']
pop_total['pop_50'] = pop_total['m_50_54'] + pop_total['m_55_59'] + pop_total['f_50_54'] + pop_total['f_55_59']
pop_total['pop_60'] = pop_total['m_60_64'] + pop_total['m_65_69'] + pop_total['f_60_64'] + pop_total['f_65_69']
pop_total['pop_over_70'] = pop_total['m_over_70'] + pop_total['f_over_70']

# 필요한 변수들만 추출

In [11]:
pop_total = pop_total[['date', 'hour', 'dong_code', 'total', 'pop_below_20', 'pop_20', 
                       'pop_30', 'pop_40', 'pop_50', 'pop_60', 'pop_over_70']]
pop_total.head()

date  hour  dong_code       total  pop_below_20     pop_20     pop_30  \
0 2018-01-01     0   11110515  14471.4116     2240.2873  1901.9084  2160.7656   
1 2018-01-01     0   11110530  23059.7992     2823.3040  3855.7544  4059.9409   
2 2018-01-01     0   11110540   4928.6410      522.2231   825.1988   692.4254   
3 2018-01-01     0   11110550  14060.9810     1774.7914  1720.8890  1733.4553   
4 2018-01-01     0   11110560  20322.3298     2807.5818  2200.1501  2287.0940   

      pop_40     pop_50     pop_60  pop_over_70  
0  2557.8714  2013.3887  1389.6512    2207.5381  
1  3975.0511  3048.1681  2131.8776    3165.6997  
2   843.7170   711.8668   570.2299     762.9801  
3  2197.9273  2334.8394  1845.8798    2453.1990  
4  3056.2584  3594.8618  2749.7658    3626.6169

# 행정동코드 변환
- 인구 데이터의 "행정동코드" 변수가 한글이 아닌 숫자로 적혀있다.
- 때문에 행정동코드와 행정동을 하나하나 매칭시켜서 변경해주도록 하겠다.
    
**<참고>**
- **'11620685':'관악구 신사동'**은 딕셔너리에서 제외시켰다.
    - 지하철 역 주소 데이터를 보면, '신사동'은 '강남구'에만 존재하기 때문이다.

In [12]:
dic = {'11110515':'종로구 청운효자동', '11110530':'종로구 사직동', '11110540':'종로구 삼청동', '11110550':'종로구 부암동', 
       '11110560':'종로구 평창동', '11110570':'종로구 무악동', '11110580':'종로구 교남동', '11110600':'종로구 가회동', 
       '11110615':'종로구 종로1.2.3.4가동', '11110630':'종로구 종로5.6가동', '11110640':'종로구 이화동', '11110650':'종로구 혜화동', 
       '11110670':'종로구 창신제1동', '11110680':'종로구 창신제2동', '11110690':'종로구 창신제3동', '11110700':'종로구 숭인제1동', 
       '11110710':'종로구 숭인제2동', 
       
       '11140520':'중구 소공동', '11140540':'중구 회현동', '11140550':'중구 명동', '11140570':'중구 필동', '11140580':'중구 장충동', 
       '11140590':'중구 광희동', '11140605':'중구 을지로동', '11140615':'중구 신당동', '11140625':'중구 다산동', '11140635':'중구 약수동', 
       '11140645':'중구 청구동', '11140650':'중구 신당제5동', '11140665':'중구 동화동', '11140670':'중구 황학동', '11140680':'중구 중림동', 
       
       '11170510':'용산구 후암동', '11170520':'용산구 용산2가동', '11170530':'용산구 남영동', '11170555':'용산구 청파동', 
       '11170560':'용산구 원효로제1동', '11170570':'용산구 원효로제2동', '11170580':'용산구 효창동', '11170590':'용산구 용문동', 
       '11170625':'용산구 한강로동', '11170630':'용산구 이촌제1동', '11170640':'용산구 이촌제2동', '11170650':'용산구 이태원제1동', 
       '11170660':'용산구 이태원제2동', '11170685':'용산구 한남동', '11170690':'용산구 서빙고동', '11170700':'용산구 보광동',
       
       '11200520':'성동구 왕십리제2동', '11200535':'성동구 왕십리도선동', '11200540':'성동구 마장동', '11200550':'성동구 사근동',
       '11200560':'성동구 행당제1동', '11200570':'성동구 행당제2동', '11200580':'성동구 응봉동', '11200590':'성동구 금호1가동',
       '11200615':'성동구 금호2.3가동', '11200620':'성동구 금호4가동', '11200645':'성동구 옥수동', '11200650':'성동구 성수1가제1동', 
       '11200660':'성동구 성수1가제2동', '11200670':'성동구 성수2가제1동', '11200690':'성동구 성수2가제3동', '11200720':'성동구 송정동', 
       '11200790':'성동구 용답동',
       
       '11215710':'광진구 화양동', '11215730':'광진구 군자동', '11215740':'광진구 중곡제1동', '11215750':'광진구 중곡제2동',
       '11215760':'광진구 중곡제3동', '11215770':'광진구 중곡제4동', '11215780':'광진구 능동', '11215810':'광진구 광장동',
       '11215820':'광진구 자양제1동', '11215830':'광진구 자양제2동', '11215840':'광진구 자양제3동', '11215847':'광진구 자양제4동',
       '11215850':'광진구 구의제1동', '11215860':'광진구 구의제2동', '11215870':'광진구 구의제3동',
       
       '11230536':'동대문구 용신동', '11230545':'동대문구 제기동', '11230560':'동대문구 전농제1동', '11230570':'동대문구 전농제2동',
       '11230600':'동대문구 답십리제1동', '11230610':'동대문구 답십리제2동', '11230650':'동대문구 장안제1동', '11230660':'동대문구 장안제2동',
       '11230705':'동대문구 청량리동', '11230710':'동대문구 회기동', '11230720':'동대문구 휘경제1동', '11230730':'동대문구 휘경제2동',
       '11230740':'동대문구 이문제1동', '11230750':'동대문구 이문제2동',
      
       '11260520':'중랑구 면목제2동', '11260540':'중랑구 면목제4동', '11260550':'중랑구 면목제5동', '11260565':'중랑구 면목본동',
       '11260570':'중랑구 면목제7동', '11260575':'중랑구 면목제3.8동', '11260580':'중랑구 상봉제1동', '11260590':'중랑구 상봉제2동', 
       '11260600':'중랑구 중화제1동', '11260610':'중랑구 중화제2동', '11260620':'중랑구 묵제1동', '11260630':'중랑구 묵제2동', 
       '11260655':'중랑구 망우본동', '11260660':'중랑구 망우제3동', '11260680':'중랑구 신내1동', '11260690':'중랑구 신내2동',
      
       '11290525':'성북구 성북동', '11290555':'성북구 삼선동', '11290575':'성북구 동선동', '11290580':'성북구 돈암제1동', 
       '11290590':'성북구 돈암제2동', '11290600':'성북구 안암동', '11290610':'성북구 보문동', '11290620':'성북구 정릉제1동', 
       '11290630':'성북구 정릉제2동', '11290640':'성북구 정릉제3동', '11290650':'성북구 정릉제4동', '11290660':'성북구 길음제1동', 
       '11290685':'성북구 길음제2동', '11290705':'성북구 종암동', '11290715':'성북구 월곡제1동', '11290725':'성북구 월곡제2동', 
       '11290760':'성북구 장위제1동', '11290770':'성북구 장위제2동', '11290780':'성북구 장위제3동', '11290810':'성북구 석관동',
       
       '11305534':'강북구 삼양동', '11305535':'강북구 미아동', '11305545':'강북구 송중동', '11305555':'강북구 송천동',
       '11305575':'강북구 삼각산동', '11305590':'강북구 번제1동', '11305600':'강북구 번제2동', '11305606':'강북구 번제3동',
       '11305610':'강북구 수유제1동', '11305620':'강북구 수유제2동', '11305630':'강북구 수유제3동', '11305645':'강북구 우이동', 
       '11305660':'강북구 인수동',
      
       '11320511':'도봉구 창제1동', '11320512':'도봉구 창제2동', '11320513':'도봉구 창제3동', '11320514':'도봉구 창제4동',
       '11320515':'도봉구 창제5동', '11320521':'도봉구 도봉제1동', '11320522':'도봉구 도봉제2동', '11320660':'도봉구 쌍문제1동',
       '11320670':'도봉구 쌍문제2동', '11320680':'도봉구 쌍문제3동', '11320681':'도봉구 쌍문제4동', '11320690':'도봉구 방학제1동',
       '11320700':'도봉구 방학제2동', '11320710':'도봉구 방학제3동',
      
       '11350560':'노원구 월계1동', '11350570':'노원구 월계2동', '11350580':'노원구 월계3동', '11350595':'노원구 공릉1동',
       '11350600':'노원구 공릉2동', '11350611':'노원구 하계1동', '11350612':'노원구 하계2동', '11350619':'노원구 중계본동',
       '11350621':'노원구 중계1동', '11350624':'노원구 중계4동', '11350625':'노원구 중계2.3동', '11350630':'노원구 상계1동', 
       '11350640':'노원구 상계2동', '11350665':'노원구 상계3.4동', '11350670':'노원구 상계5동', '11350695':'노원구 상계6.7동', 
       '11350700':'노원구 상계8동', '11350710':'노원구 상계9동', '11350720':'노원구 상계10동',
      
       '11380510':'은평구 녹번동', '11380520':'은평구 불광제1동', '11380530':'은평구 불광제2동', '11380551':'은평구 갈현제1동',
       '11380552':'은평구 갈현제2동', '11380560':'은평구 구산동', '11380570':'은평구 대조동', '11380580':'은평구 응암제1동',
       '11380590':'은평구 응암제2동', '11380600':'은평구 응암제3동', '11380625':'은평구 역촌동', '11380631':'은평구 신사제1동',
       '11380632':'은평구 신사제2동', '11380640':'은평구 증산동', '11380650':'은평구 수색동', '11380690':'은평구 진관동',
      
       '11410520':'서대문구 천연동', '11410555':'서대문구 북아현동', '11410565':'서대문구 충현동', '11410585':'서대문구 신촌동',
       '11410615':'서대문구 연희동', '11410620':'서대문구 홍제제1동', '11410640':'서대문구 홍제제3동', '11410655':'서대문구 홍제제2동',
       '11410660':'서대문구 홍은제1동', '11410685':'서대문구 홍은제2동', '11410690':'서대문구 남가좌제1동', '11410700':'서대문구 남가좌제2동',
       '11410710':'서대문구 북가좌제1동', '11410720':'서대문구 북가좌제2동',
      
       '11440555':'마포구 아현동', '11440565':'마포구 공덕동', '11440585':'마포구 도화동', '11440590':'마포구 용강동', 
       '11440600':'마포구 대흥동', '11440610':'마포구 염리동', '11440630':'마포구 신수동', '11440655':'마포구 서강동',
       '11440660':'마포구 서교동', '11440680':'마포구 합정동', '11440690':'마포구 망원제1동', '11440700':'마포구 망원제2동',
       '11440710':'마포구 연남동', '11440720':'마포구 성산제1동', '11440730':'마포구 성산제2동', '11440740':'마포구 상암동',
      
       '11470510':'양천구 목1동', '11470520':'양천구 목2동', '11470530':'양천구 목3동', '11470540':'양천구 목4동',
       '11470550':'양천구 목5동', '11470560':'양천구 신월1동', '11470570':'양천구 신월2동', '11470580':'양천구 신월3동',
       '11470590':'양천구 신월4동', '11470600':'양천구 신월5동', '11470610':'양천구 신월6동', '11470611':'양천구 신월7동',
       '11470620':'양천구 신정1동', '11470630':'양천구 신정2동', '11470640':'양천구 신정3동', '11470650':'양천구 신정4동',
       '11470670':'양천구 신정6동', '11470680':'양천구 신정7동',
      
       '11500510':'강서구 염창동', '11500520':'강서구 등촌제1동', '11500530':'강서구 등촌제2동', '11500535':'강서구 등촌제3동',
       '11500540':'강서구 화곡제1동', '11500550':'강서구 화곡제2동', '11500560':'강서구 화곡제3동', '11500570':'강서구 화곡제4동',
       '11500590':'강서구 화곡본동', '11500591':'강서구 화곡제6동', '11500593':'강서구 화곡제8동', '11500603':'강서구 가양제1동',
       '11500604':'강서구 가양제2동', '11500605':'강서구 가양제3동', '11500611':'강서구 발산제1동', '11500615':'강서구 우장산동',
       '11500620':'강서구 공항동', '11500630':'강서구 방화제1동', '11500640':'강서구 방화제2동', '11500641':'강서구 방화제3동',
      
       '11530510':'구로구 신도림동', '11530520':'구로구 구로제1동', '11530530':'구로구 구로제2동', '11530540':'구로구 구로제3동',
       '11530550':'구로구 구로제4동', '11530560':'구로구 구로제5동', '11530595':'구로구 가리봉동', '11530720':'구로구 고척제1동',
       '11530730':'구로구 고척제2동', '11530740':'구로구 개봉제1동', '11530750':'구로구 개봉제2동', '11530760':'구로구 개봉제3동',
       '11530770':'구로구 오류제1동', '11530780':'구로구 오류제2동', '11530790':'구로구 수궁동',
      
       '11545510':'금천구 가산동', '11545610':'금천구 독산제1동', '11545620':'금천구 독산제2동', '11545630':'금천구 독산제3동',
       '11545640':'금천구 독산제4동', '11545670':'금천구 시흥제1동', '11545680':'금천구 시흥제2동', '11545690':'금천구 시흥제3동',
       '11545700':'금천구 시흥제4동', '11545710':'금천구 시흥제5동',
      
       '11560515':'영등포구 영등포본동', '11560535':'영등포구 영등포동', '11560540':'영등포구 여의동', '11560550':'영등포구 당산제1동',
       '11560560':'영등포구 당산제2동', '11560585':'영등포구 도림동', '11560605':'영등포구 문래동', '11560610':'영등포구 양평제1동',
       '11560620':'영등포구 양평제2동', '11560630':'영등포구 신길제1동', '11560650':'영등포구 신길제3동', '11560660':'영등포구 신길제4동',
       '11560670':'영등포구 신길제5동', '11560680':'영등포구 신길제6동', '11560690':'영등포구 신길제7동', '11560700':'영등포구 대림제1동',
       '11560710':'영등포구 대림제2동', '11560720':'영등포구 대림제3동',
      
       '11590510':'동작구 노량진제1동', '11590520':'동작구 노량진제2동', '11590530':'동작구 상도제1동', '11590540':'동작구 상도제2동',
       '11590550':'동작구 상도제3동', '11590560':'동작구 상도제4동', '11590605':'동작구 흑석동', '11590620':'동작구 사당제1동',
       '11590630':'동작구 사당제2동', '11590640':'동작구 사당제3동', '11590650':'동작구 사당제4동', '11590651':'동작구 사당제5동',
       '11590660':'동작구 대방동', '11590670':'동작구 신대방제1동', '11590680':'동작구 신대방제2동',
      
       '11620525':'관악구 보라매동', '11620545':'관악구 청림동', '11620565':'관악구 성현동', '11620575':'관악구 행운동',
       '11620585':'관악구 낙성대동', '11620595':'관악구 청룡동', '11620605':'관악구 은천동', '11620615':'관악구 중앙동',
       '11620625':'관악구 인헌동', '11620630':'관악구 남현동', '11620645':'관악구 서원동', '11620655':'관악구 신원동',
       '11620665':'관악구 서림동', '11620695':'관악구 신림동', '11620715':'관악구 난향동', '11620725':'관악구 조원동', 
       '11620735':'관악구 대학동', '11620745':'관악구 삼성동', '11620765':'관악구 미성동', '11620775':'관악구 난곡동',
      
       '11650510':'서초구 서초1동', '11650520':'서초구 서초2동', '11650530':'서초구 서초3동', '11650531':'서초구 서초4동',
       '11650540':'서초구 잠원동', '11650550':'서초구 반포본동', '11650560':'서초구 반포1동', '11650570':'서초구 반포2동',
       '11650580':'서초구 반포3동', '11650581':'서초구 반포4동', '11650590':'서초구 방배본동', '11650600':'서초구 방배1동',
       '11650610':'서초구 방배2동', '11650620':'서초구 방배3동', '11650621':'서초구 방배4동', '11650651':'서초구 양재1동',
       '11650652':'서초구 양재2동', '11650660':'서초구 내곡동',
      
       '11680510':'강남구 신사동', '11680521':'강남구 논현1동', '11680531':'강남구 논현2동', '11680545':'강남구 압구정동',
       '11680565':'강남구 청담동', '11680580':'강남구 삼성1동', '11680590':'강남구 삼성2동', '11680600':'강남구 대치1동',
       '11680610':'강남구 대치2동', '11680630':'강남구 대치4동', '11680640':'강남구 역삼1동', '11680650':'강남구 역삼2동',
       '11680655':'강남구 도곡1동', '11680656':'강남구 도곡2동', '11680660':'강남구 개포1동', '11680670':'강남구 개포2동',
       '11680690':'강남구 개포4동', '11680700':'강남구 세곡동', '11680720':'강남구 일원본동', '11680730':'강남구 일원1동',
       '11680740':'강남구 일원2동', '11680750':'강남구 수서동',
      
       '11710510':'송파구 풍납1동', '11710520':'송파구 풍납2동', '11710531':'송파구 거여1동', '11710532':'송파구 거여2동',
       '11710540':'송파구 마천1동', '11710550':'송파구 마천2동', '11710561':'송파구 방이1동', '11710562':'송파구 방이2동',
       '11710566':'송파구 오륜동', '11710570':'송파구 오금동', '11710580':'송파구 송파1동', '11710590':'송파구 송파2동',
       '11710600':'송파구 석촌동', '11710610':'송파구 삼전동', '11710620':'송파구 가락본동', '11710631':'송파구 가락1동',
       '11710632':'송파구 가락2동', '11710641':'송파구 문정1동', '11710642':'송파구 문정2동', '11710646':'송파구 장지동',
       '11710647':'송파구 위례동', '11710650':'송파구 잠실본동', '11710670':'송파구 잠실2동', '11710680':'송파구 잠실3동', 
       '11710690':'송파구 잠실4동', '11710710':'송파구 잠실6동', '11710720':'송파구 잠실7동',
      
       '11740515':'강동구 강일동', '11740520':'강동구 상일동', '11740530':'강동구 명일제1동', '11740540':'강동구 명일제2동',
       '11740550':'강동구 고덕제1동', '11740560':'강동구 고덕제2동', '11740570':'강동구 암사제1동', '11740580':'강동구 암사제2동',
       '11740590':'강동구 암사제3동', '11740600':'강동구 천호제1동', '11740610':'강동구 천호제2동', '11740620':'강동구 천호제3동',
       '11740640':'강동구 성내제1동', '11740650':'강동구 성내제2동', '11740660':'강동구 성내제3동', '11740685':'강동구 길동',
       '11740690':'강동구 둔촌제1동', '11740700':'강동구 둔촌제2동'}

In [13]:
dic_df = pd.DataFrame.from_dict(dic, orient = 'index')
dic_df = dic_df.reset_index()
dic_df.rename(columns = {'index': 'dong_code', 0: 'dong'}, inplace = True)
dic_df

dong_code       dong
0    11110515  종로구 청운효자동
1    11110530    종로구 사직동
2    11110540    종로구 삼청동
3    11110550    종로구 부암동
4    11110560    종로구 평창동
..        ...        ...
418  11740650  강동구 성내제2동
419  11740660  강동구 성내제3동
420  11740685     강동구 길동
421  11740690  강동구 둔촌제1동
422  11740700  강동구 둔촌제2동

[423 rows x 2 columns]

- dong 변수를 "종로구_청운효자동" 이런 식으로 변경해주겠다.

In [14]:
result = []
for i in range(len(dic_df)):
    dong_split = dic_df['dong'][i].split()
    result.append('_'.join(dong_split))
    
result

['종로구_청운효자동',
 '종로구_사직동',
 '종로구_삼청동',
 '종로구_부암동',
 '종로구_평창동',
 '종로구_무악동',
 '종로구_교남동',
 '종로구_가회동',
 '종로구_종로1.2.3.4가동',
 '종로구_종로5.6가동',
 '종로구_이화동',
 '종로구_혜화동',
 '종로구_창신제1동',
 '종로구_창신제2동',
 '종로구_창신제3동',
 '종로구_숭인제1동',
 '종로구_숭인제2동',
 '중구_소공동',
 '중구_회현동',
 '중구_명동',
 '중구_필동',
 '중구_장충동',
 '중구_광희동',
 '중구_을지로동',
 '중구_신당동',
 '중구_다산동',
 '중구_약수동',
 '중구_청구동',
 '중구_신당제5동',
 '중구_동화동',
 '중구_황학동',
 '중구_중림동',
 '용산구_후암동',
 '용산구_용산2가동',
 '용산구_남영동',
 '용산구_청파동',
 '용산구_원효로제1동',
 '용산구_원효로제2동',
 '용산구_효창동',
 '용산구_용문동',
 '용산구_한강로동',
 '용산구_이촌제1동',
 '용산구_이촌제2동',
 '용산구_이태원제1동',
 '용산구_이태원제2동',
 '용산구_한남동',
 '용산구_서빙고동',
 '용산구_보광동',
 '성동구_왕십리제2동',
 '성동구_왕십리도선동',
 '성동구_마장동',
 '성동구_사근동',
 '성동구_행당제1동',
 '성동구_행당제2동',
 '성동구_응봉동',
 '성동구_금호1가동',
 '성동구_금호2.3가동',
 '성동구_금호4가동',
 '성동구_옥수동',
 '성동구_성수1가제1동',
 '성동구_성수1가제2동',
 '성동구_성수2가제1동',
 '성동구_성수2가제3동',
 '성동구_송정동',
 '성동구_용답동',
 '광진구_화양동',
 '광진구_군자동',
 '광진구_중곡제1동',
 '광진구_중곡제2동',
 '광진구_중곡제3동',
 '광진구_중곡제4동',
 '광진구_능동',
 '광진구_광장동',
 '광진구_자양제1동',
 '광진구_자양제2동',
 '광진구_자양제3동',
 '광진구_자양제4동

In [15]:
dic_df['dong'] = result
dic_df

dong_code       dong
0    11110515  종로구_청운효자동
1    11110530    종로구_사직동
2    11110540    종로구_삼청동
3    11110550    종로구_부암동
4    11110560    종로구_평창동
..        ...        ...
418  11740650  강동구_성내제2동
419  11740660  강동구_성내제3동
420  11740685     강동구_길동
421  11740690  강동구_둔촌제1동
422  11740700  강동구_둔촌제2동

[423 rows x 2 columns]

- 인구 데이터 셋(pop_total)과 수정된 행정동코드 데이터 셋(dic_df)의 'dong_code' 변수 타입이 서로 다르다.
- 때문에 dic_df 데이터 셋의 'dong_code' 변수를 int형으로 변경해주겠다.

In [16]:
pop_total.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7306368 entries, 0 to 7306367
Data columns (total 11 columns):
 #   Column        Dtype         
---  ------        -----         
 0   date          datetime64[ns]
 1   hour          int64         
 2   dong_code     int64         
 3   total         float64       
 4   pop_below_20  float64       
 5   pop_20        float64       
 6   pop_30        float64       
 7   pop_40        float64       
 8   pop_50        float64       
 9   pop_60        float64       
 10  pop_over_70   float64       
dtypes: datetime64[ns](1), float64(8), int64(2)
memory usage: 613.2 MB


In [17]:
dic_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 423 entries, 0 to 422
Data columns (total 2 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   dong_code  423 non-null    object
 1   dong       423 non-null    object
dtypes: object(2)
memory usage: 6.7+ KB


In [18]:
dic_df['dong_code'] = pd.to_numeric(dic_df['dong_code'])
dic_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 423 entries, 0 to 422
Data columns (total 2 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   dong_code  423 non-null    int64 
 1   dong       423 non-null    object
dtypes: int64(1), object(1)
memory usage: 6.7+ KB


- pop_total 데이터 셋에는 'dong_code'가 424개가 존재하는데, dic_df 데이터 셋에는 'dong_code'가 423개 존재한다.
    - 최종 inner join한 데이터 셋을 통해, 1개 차이로 인해 len이 불일치함을 확인하였다.
- 따라서 이 부분을 해결해주고 inner join을 수행해주도록 하겠다.
    - 해결 방안은 직접 비교해서 dic_df 데이터 셋에 미존재하는 **"11710647(00)"** 행정동을 직접 찾아서 넣어주어야 한다.
    - 구글링을 해 본 결과, **"11710647(00)"**에 해당하는 행정동은 **"위례동"**임을 확인했다.
    - 그래서 dic(행정동코드 변환 딕셔너리)에 **```'11710647':'위례동'```** 코드를 넣어주었다.
    
**<참고>**
- 위 사항들을 확인해보기 위한 코드는 모두 주석 처리 해두었다.

In [19]:
# pop_total_final = pd.merge(pop_total, dic_df, how = 'inner', on = 'dong_code')
# pop_total_final

In [20]:
# df1 = pd.DataFrame(set(pop_total['dong_code']))
# df2 = pd.DataFrame(set(dic_df['dong_code']))

# df1.rename(columns = {0:'dong_code'}, inplace = True)
# df2.rename(columns = {0:'dong_code'}, inplace = True)

# check1 = df1.sort_values(by = 'dong_code')
# check2 = df2.sort_values(by = 'dong_code')

In [21]:
# # c_total 데이터 셋을 50개의 row 씩 잘라서 눈으로 확인
# # 그 결과, dic_df 데이터 셋에 "11710647" 행정동코드가 미존재함을 확인!!
# c_total = pd.concat([check1, check2], axis = 1)

In [22]:
# pop_total[pop_total['dong_code'] == 11710647]

- 이제 pop_total 데이터 셋과 dic_df 데이터 셋을 'dong_code' 변수를 조인키로 사용하여 inner join 해주겠다.

In [23]:
pop_total_final = pd.merge(pop_total, dic_df, how = 'inner', on = 'dong_code')
pop_total_final

date  hour  dong_code       total  pop_below_20     pop_20  \
0       2018-01-01     0   11110515  14471.4116     2240.2873  1901.9084   
1       2018-01-01     1   11110515  14476.3838     2192.6338  1839.7302   
2       2018-01-01     2   11110515  14624.7353     2179.0636  1883.6808   
3       2018-01-01     3   11110515  14667.7190     2187.9826  1884.9999   
4       2018-01-01     4   11110515  14724.5581     2203.3030  1894.6867   
...            ...   ...        ...         ...           ...        ...   
7289131 2019-12-31    19   11740700  28966.0305     4402.7464  2984.5773   
7289132 2019-12-31    20   11740700  29495.9850     4460.1392  3123.6626   
7289133 2019-12-31    21   11740700  30131.0225     4616.2185  3210.7351   
7289134 2019-12-31    22   11740700  31575.9329     5043.9562  3474.2488   
7289135 2019-12-31    23   11740700  32974.4339     5135.1512  3596.6190   

            pop_30     pop_40     pop_50     pop_60  pop_over_70       dong  
0        2160.7656  2557.8714  2013.3887  1389.6512    2207.5381  종로구_청운효자동  
1        2156.5959  2608.1911  2055.7146  1399.1206    2224.3969  종로구_청운효자동  
2        2231.6691  2614.5692  2098.7156  1386.8433    2230.1943  종로구_청운효자동  
3        2196.4550  2634.9925  2100.9159  1399.4056    2262.9676  종로구_청운효자동  
4        2206.1891  2645.6112  2122.2458  1395.9627    2256.5594  종로구_청운효자동  
...            ...        ...        ...        ...          ...        ...  
7289131  3907.6240  4294.0431  4656.8790  4323.5792    4396.5802  강동구_둔촌제2동  
7289132  3978.6890  4387.9537  4781.8335  4324.7054    4439.0004  강동구_둔촌제2동  
7289133  4093.1960  4452.0349  4850.2092  4426.8812    4481.7475  강동구_둔촌제2동  
7289134  4300.4638  4637.4441  5022.1798  4576.0351    4521.6059  강동구_둔촌제2동  
7289135  4412.7418  4875.9918  5253.7418  4830.2941    4869.8948  강동구_둔촌제2동  

[7289136 rows x 12 columns]

In [24]:
pop_total_final.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 7289136 entries, 0 to 7289135
Data columns (total 12 columns):
 #   Column        Dtype         
---  ------        -----         
 0   date          datetime64[ns]
 1   hour          int64         
 2   dong_code     int64         
 3   total         float64       
 4   pop_below_20  float64       
 5   pop_20        float64       
 6   pop_30        float64       
 7   pop_40        float64       
 8   pop_50        float64       
 9   pop_60        float64       
 10  pop_over_70   float64       
 11  dong          object        
dtypes: datetime64[ns](1), float64(8), int64(2), object(1)
memory usage: 723.0+ MB


# 인구수 관련 변수들 데이터 타입 변경
- 소수점으로 되어 있는 인구수 관련 변수들을 모두 '올림' 처리하여 정수로 만들어주겠다.

In [25]:
var_list = ['total', 'pop_below_20', 'pop_20', 'pop_30', 'pop_40', 'pop_50', 'pop_60', 'pop_over_70']
pop_total_final[var_list] = np.ceil(pop_total_final[var_list])

In [26]:
pop_total_final.head()

date  hour  dong_code    total  pop_below_20  pop_20  pop_30  pop_40  \
0 2018-01-01     0   11110515  14472.0        2241.0  1902.0  2161.0  2558.0   
1 2018-01-01     1   11110515  14477.0        2193.0  1840.0  2157.0  2609.0   
2 2018-01-01     2   11110515  14625.0        2180.0  1884.0  2232.0  2615.0   
3 2018-01-01     3   11110515  14668.0        2188.0  1885.0  2197.0  2635.0   
4 2018-01-01     4   11110515  14725.0        2204.0  1895.0  2207.0  2646.0   

   pop_50  pop_60  pop_over_70       dong  
0  2014.0  1390.0       2208.0  종로구_청운효자동  
1  2056.0  1400.0       2225.0  종로구_청운효자동  
2  2099.0  1387.0       2231.0  종로구_청운효자동  
3  2101.0  1400.0       2263.0  종로구_청운효자동  
4  2123.0  1396.0       2257.0  종로구_청운효자동

# 조인키 생성
- 지하철 데이터 셋을 기준으로 인구 데이터 셋과 기상 데이터 셋을 결합시켜 줄 것이므로, 이를 위한 조인키가 필요하다.
- 인구 데이터의 'dong' 변수의 경우, "종로구_청운효자동" 이런 식으로 자치구와 행정동이 같이 입력되어 있다.
- 따라서 다음과 같이 조인키를 생성해주겠다.
    1. 앞서 했던 방법대로 먼저 문자열을 결합해준다.
        - ex) "2018-01-01_종로구_청운효자동"
    2. 만들어진 조인키 변수에서 자치구에 해당 되는 부분은 제거해준다.
        - ex) "2018-01-01_청운효자동"

In [27]:
# datetime to string
pop_total_final['date'] = pop_total_final['date'].astype(str)

# 문자열 결합
pop_total_final['pop_join_key'] = pop_total_final['date'] + '_' + pop_total_final['dong']
# '_'를 기준으로 문자열을 잘라서 리스트에 담기
pop_join_key_list = pop_total_final['pop_join_key'].str.split('_').to_list()

In [28]:
# '자치구' 부분을 제외해 준 뒤에 문자열 재결합
result = []
for i in tqdm(pop_join_key_list):
    result.append('_'.join([i[0], i[2]]))

# 지하철 데이터와 결합할 때 사용할 조인키
pop_total_final['pop_join_key'] = result
pop_total_final['pop_join_key']

0          2018-01-01_청운효자동
1          2018-01-01_청운효자동
2          2018-01-01_청운효자동
3          2018-01-01_청운효자동
4          2018-01-01_청운효자동
                 ...       
7289131    2019-12-31_둔촌제2동
7289132    2019-12-31_둔촌제2동
7289133    2019-12-31_둔촌제2동
7289134    2019-12-31_둔촌제2동
7289135    2019-12-31_둔촌제2동
Name: pop_join_key, Length: 7289136, dtype: object

# 불필요한 변수들 제거
- dong_code, dong 변수는 분석에 사용하지 않을 것이므로 제거해주겠다.

In [29]:
pop_total_final.columns

Index(['date', 'hour', 'dong_code', 'total', 'pop_below_20', 'pop_20',
       'pop_30', 'pop_40', 'pop_50', 'pop_60', 'pop_over_70', 'dong',
       'pop_join_key'],
      dtype='object')

In [30]:
pop_total_final.drop(['dong_code', 'dong'], axis = 1, inplace = True)

In [31]:
pop_total_final

date  hour    total  pop_below_20  pop_20  pop_30  pop_40  \
0        2018-01-01     0  14472.0        2241.0  1902.0  2161.0  2558.0   
1        2018-01-01     1  14477.0        2193.0  1840.0  2157.0  2609.0   
2        2018-01-01     2  14625.0        2180.0  1884.0  2232.0  2615.0   
3        2018-01-01     3  14668.0        2188.0  1885.0  2197.0  2635.0   
4        2018-01-01     4  14725.0        2204.0  1895.0  2207.0  2646.0   
...             ...   ...      ...           ...     ...     ...     ...   
7289131  2019-12-31    19  28967.0        4403.0  2985.0  3908.0  4295.0   
7289132  2019-12-31    20  29496.0        4461.0  3124.0  3979.0  4388.0   
7289133  2019-12-31    21  30132.0        4617.0  3211.0  4094.0  4453.0   
7289134  2019-12-31    22  31576.0        5044.0  3475.0  4301.0  4638.0   
7289135  2019-12-31    23  32975.0        5136.0  3597.0  4413.0  4876.0   

         pop_50  pop_60  pop_over_70      pop_join_key  
0        2014.0  1390.0       2208.0  2018-01-01_청운효자동  
1        2056.0  1400.0       2225.0  2018-01-01_청운효자동  
2        2099.0  1387.0       2231.0  2018-01-01_청운효자동  
3        2101.0  1400.0       2263.0  2018-01-01_청운효자동  
4        2123.0  1396.0       2257.0  2018-01-01_청운효자동  
...         ...     ...          ...               ...  
7289131  4657.0  4324.0       4397.0  2019-12-31_둔촌제2동  
7289132  4782.0  4325.0       4440.0  2019-12-31_둔촌제2동  
7289133  4851.0  4427.0       4482.0  2019-12-31_둔촌제2동  
7289134  5023.0  4577.0       4522.0  2019-12-31_둔촌제2동  
7289135  5254.0  4831.0       4870.0  2019-12-31_둔촌제2동  

[7289136 rows x 11 columns]

In [32]:
pop_total_final.isnull().sum()

date            0
hour            0
total           0
pop_below_20    0
pop_20          0
pop_30          0
pop_40          0
pop_50          0
pop_60          0
pop_over_70     0
pop_join_key    0
dtype: int64

In [33]:
pop_total_final.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 7289136 entries, 0 to 7289135
Data columns (total 11 columns):
 #   Column        Dtype  
---  ------        -----  
 0   date          object 
 1   hour          int64  
 2   total         float64
 3   pop_below_20  float64
 4   pop_20        float64
 5   pop_30        float64
 6   pop_40        float64
 7   pop_50        float64
 8   pop_60        float64
 9   pop_over_70   float64
 10  pop_join_key  object 
dtypes: float64(8), int64(1), object(2)
memory usage: 667.3+ MB


# 최종 생성된 데이터 셋을 csv 파일로 내보내기

In [34]:
new_filepath = 'C:/Users/Brian/Desktop/Data Science/Projects/Playdata/dataset/made_by_전처리/train'

In [35]:
pop_total_final.to_csv('{}/pop_final.csv'.format(new_filepath), index = False)